### ISSUE TITLE
chart.svg.querySelectorAll('.ct-vertical') leads to error

### ISSUE THREAD

**Issue by gsklee** [chartist-js/chartist#261]
I have trouble selecting particular elements, ie. whenever I do `chart.svg.querySelectorAll('.ct-vertical')` it will leads to the following error:

```
Uncaught InvalidCharacterError: Failed to execute 'createElementNS' on 'Document': The qualified name provided ('[object HTMLSpanElement]') contains the invalid name-start character '['.
```

Demo: http://jsbin.com/yatodamexo/2/edit?html,js,output

**Comment by gionkunz** [comment]
Hey there

Thanks for the issue! This is clearly a bug. The problem is with the foreignObject labels. Only the foreign objects are SVG elements but the labels inside are spans. Chartist.Svg was designed to only handle SVGElement and if it's not an SVG element it will assume you wanted to pass a name for creating a new element. The stack trace is:

```
Svg (chartist.js:1374)
constr (chartist.js:1115)
SvgList (chartist.js:1905)
constr (chartist.js:1115)
querySelectorAll (chartist.js:1490)
```

As the querySelectorAll creates a new SvgList that creates new Svg elements based on the Node list selected by `Element.querySelectorAll`. 

The Problem is this part: https://github.com/gionkunz/chartist-js/blob/v0.7.3/src/scripts/svg.js#L33-L37

As you can see the `name` parameter is checked if this element is a SVGElement, and if so it only creates a wrapper element. Otherwise it will create a new element in the DOM. 

If you force Chartist to use SVG text instead of foreign objects the error you have seen disappears: http://jsbin.com/pemahi/1/edit

I think there are two approaches to fix this issue. 

A: Change the CSS selectors and make the actual foreign Object contain the classes and make the styles to also apply to span found in the foreign Object. This would be nice since we are reducing out concerns to SVG only.
B: Change the SVG module to something more generic like a DOM module that does not freak out with regular HTML Elements.
C: Quick fix by just changing the SVGElement detection to Element and close eyes from any design issues that this may reveal

Thoughts?

**Comment by gsklee** [comment]
@gionkunz since you choose to include `foreignObject` support in the first place I'd say (B) seems to be the most generic solution which will enhance the lib's capability to manipulate embedded HTML elements; meanwhile I can get around this issue using the workaround found in your demo ;-)

**Comment by gionkunz** [comment]
@gsklee I've considered to go down route B, but it brings a lot of issues with backwards compatibility. I've implemented the quick fix and will open a ticket for B to be implemented for 1.0.0

**Comment by gionkunz** [comment]
Fixed in 50e9724

**Comment by gsklee** [comment]
@gionkunz thanks!
